### PR TITLE
Add switch for motion detection

### PR DESCRIPTION
**Pull request by SamMousa**
This depends on my PR here: https://github.com/cgarwood/python-fullykiosk/pull/10

I've tested it to be working on my installation without modifying the underlying python library (harder to reach from vs-code plugin). To do that I just adapted the code from this PR to call existing function in `python-fullykiosk` directly.

So instead of calling `enableMotionDetection` etc I did this:
```python
class FullyKioskMotionDetectionSwitch(FullySwitch):
    """Representation of a Fully Kiosk Browser kiosk lock switch."""

    def __init__(self, hass, coordinator):
        """Intialize the kiosk lock switch."""
        super().__init__(hass, coordinator)
        self._name = f"{coordinator.data['deviceName']} Motion Detection"
        self._unique_id = f"{coordinator.data['deviceID']}-motion-detection"

    @property
    def is_on(self):
        """Return if motion detection is on."""
        return self.coordinator.data["settings"]["motionDetection"]

    async def async_turn_on(self, **kwargs):
        """Turn on kiosk lock."""
        await self.coordinator.fully.setConfigurationBool("motionDetection", True)
        await self.coordinator.async_refresh()

    async def async_turn_off(self, **kwargs):
        """Turn off kiosk lock."""
        await self.coordinator.fully.setConfigurationBool("motionDetection", False)
        await self.coordinator.async_refresh()

```

And instead of calling `getSettings` in `coordinator.py` I did this:
```python
async def _async_update_data(self):
        """Update data via library."""
        try:
            with timeout(15):
                """Get device info and settings in parallel"""
                result = await asyncio.gather(self.fully.getDeviceInfo(), self.fully.sendCommand("listSettings"))
                """Store settings under settings key in data"""
                result[0]["settings"] = result[1]
                return result[0]
        except (FullyKioskError, ClientConnectorError) as error:
            raise UpdateFailed(error) from error
```

Again I don't speak python so check thoroughly!

This should also provide a simple framework for adding other switches for settings.

### CODE REVIEW

**Comment by SamMousa** [comment]
Does this look good @cgarwood?

**Comment by bcutter** [comment]
> 
> 
> Does this look good @cgarwood?

Just want to ask the same question @cgarwood 2,5 months later :-)

**Comment by cgarwood** [comment]
😬 so far behind on projects.. just merged your other PR and did a new release of the library, 0.0.9. Please update the requirements in manifest.json and give it a quick test then I think we're good to merge.

**Comment by SamMousa** [comment]
@cgarwood Done!

**Comment by chpego** [comment]
Hi @SamMousa , did you try to fix the format of your commit ?

**Comment by SamMousa** [comment]
Not sure what you mean? I thought this one was done, but haven't looked at it in a long while!

**Comment by chpego** [comment]
> Not sure what you mean? I thought this one was done, but haven't looked at it in a long while!

the check formatting action failed on some parts of the code you pushed :
https://github.com/cgarwood/homeassistant-fullykiosk/runs/4289157913?check_suite_focus=true#step:4:61

File: custom_components/fullykiosk/coordinator.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 """Provides the The Fully Kiosk Browser DataUpdateCoordinator."""
 import logging
+import asyncio
```
Comment (chpego) [inline custom_components/fullykiosk/coordinator.py:3]: reverse this two lines

File: custom_components/fullykiosk/coordinator.py [hunk 2]
```diff
@@ -35,6 +36,10 @@ async def _async_update_data(self):
         """Update data via library."""
         try:
             with timeout(15):
-                return await self.fully.getDeviceInfo()
+                """Get device info and settings in parallel"""
+                result = await asyncio.gather(self.fully.getDeviceInfo(), self.fully.getSettings())
```
Comment (chpego) [inline custom_components/fullykiosk/coordinator.py:40]: try this:

```
result = await asyncio.gather(
             self.fully.getDeviceInfo(), self.fully.getSettings()
)
```
instead in one line

**Comment by SamMousa** [comment]
Yeah will do when I have time!

**Review by cgarwood — APPROVED** [review]
Had some free time so I committed the code formatting changes 🙂

HACS validation test failure is unrelated to this PR

**Pull request merged**